controllers/useraccount/useraccount_controller_test.go

```diff
@@ -671,104 +702,203 @@ func TestReconcile(t *testing.T) {
+				require.Error(t, err)
+				assert.True(t, apierros.IsNotFound(err))
+				useraccount.AssertThatUserAccount(t, userAcc.Name, cl).HasConditions(terminating("deleting user/identity"))
+
+				t.Run("second reconcile deletes user", func(t *testing.T) {
+					// when
+					res, err = r.Reconcile(context.TODO(), req)
+
+					// then
+					assert.Equal(t, reconcile.Result{}, res)
+					require.NoError(t, err)
+
+					useraccount.AssertThatUserAccount(t, req.Name, cl).
+						HasConditions(notReady("Terminating", "deleting user/identity"))
+
+					// Check that the associated user has been deleted
+					// when reconciling the useraccount with a deletion timestamp
+					assertUserNotFound(t, r, userAcc)
+					useraccount.AssertThatUserAccount(t, userAcc.Name, cl).HasConditions(terminating("deleting user/identity"))
+
+					t.Run("third reconcile deletes NSTemplateSet", func(t *testing.T) {
+						// when
+						res, err = r.Reconcile(context.TODO(), req)
+
+						// then
+						assert.Equal(t, reconcile.Result{}, res)
+						require.NoError(t, err)
+
+						// Check that the user account finalizer has been removed
+						// when reconciling the useraccount with a deletion timestamp
+						useraccount.AssertThatUserAccount(t, username, r.Client).
+							HasConditions(terminating("deleting NSTemplateSet"))
+
+						t.Run("fourth reconcile removes finalizer", func(t *testing.T) {
+							// when
+							res, err = r.Reconcile(context.TODO(), req)
+
+							// then
+							assert.Equal(t, reconcile.Result{}, res)
+							require.NoError(t, err)
+
+							// Check that the user account finalizer has been removed
+							// when reconciling the useraccount with a deletion timestamp
+							useraccount.AssertThatUserAccount(t, username, r.Client).
+								HasNoFinalizer()
+							require.Equal(t, 7, *mockCallsCounter)
+						})
+					})
+				})
+			})
+		})
+
+		t.Run("without NSTemlateSet", func(t *testing.T) {
+			// given
+
+			// when the member operator secret exists and has a che admin user configured then che user deletion is enabled
+			cfg := commonconfig.NewMemberOperatorConfigWithReset(t,
+				testconfig.Che().
+					UserDeletionEnabled(true).
+					KeycloakRouteName("keycloak").
+					Secret().
+					Ref("test-secret").
+					CheAdminUsernameKey("che.admin.username").
+					CheAdminPasswordKey("che.admin.password"))
+
+			mockCallsCounter := new(int)
+			defer gock.OffAll()
+			gockTokenSuccess(mockCallsCounter)
+			gockFindUserTimes(username, 2, mockCallsCounter)
+			gockFindUserNoBody(username, 404, mockCallsCounter)
+			gockDeleteUser(204, mockCallsCounter)
+
+			memberOperatorSecret := newSecretWithCheAdminCreds()
+			userAcc := newUserAccount(username, userID, withoutNSTemplateSet())
+			util.AddFinalizer(userAcc, toolchainv1alpha1.FinalizerName)
+			r, req, cl, _ := prepareReconcile(t, username, cfg, userAcc, preexistingUser, preexistingIdentity, memberOperatorSecret, cheRoute(true), keycloackRoute(true))
```

Comment:
you forgot to add `preexistingNsTmplSet` - that's why it's passing
```suggestion
			r, req, cl, _ := prepareReconcile(t, username, cfg, userAcc, preexistingUser, preexistingIdentity, preexistingNsTmplSet, memberOperatorSecret, cheRoute(true), keycloackRoute(true))
```